### PR TITLE
Add ITK-VTK-Viewer example

### LINKED ISSUE #3: Add CI testing

### LINKED ISSUE #7: Output transfer function for VTK.js consumption
- [ ] VTK.js volume renderer example with TfEditor
- [ ] Add utility function converting TfEditor Points to VTK.js transfer function

### PR DESCRIPTION
**Pull request by PaulHax**
closes #3 
closes #7

### CODE REVIEW

File: examples/itk-vtk-viewer/index.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        left: 0;
+        overflow: hidden;
+        background: black;
+        margin: 0;
+        padding: 0;
+      "
+    ></div>
+    <script
+      type="text/javascript"
+      src="https://unpkg.com/itk-vtk-viewer@11.11.2/dist/itkVtkViewerCDN.js"
```
Comment (thewtex) [inline examples/itk-vtk-viewer/index.html]: With a recent update, 11,12.1, there is a major update to the latest vtk.js, itk-wasm, and support for `unpkg.com` -> `jsdelivr.com/npm` (faster and more reliable).

Comment (PaulHax) [reply examples/itk-vtk-viewer/index.html]: Upgraded. It does feel faster!

Comment (thewtex) [reply examples/itk-vtk-viewer/index.html]: :+1:

**Review by thewtex — CHANGES_REQUESTED** [review]
Getting there:

![image](https://user-images.githubusercontent.com/25432/161153234-d9f370f6-f931-4d62-8e06-e6872405830c.png)


In local testing, noticed:

- The itk-vtk-viewer example is not using this transfer function editor. This is a priority.
- The vtk-js example needs `npm i --save-dev vite`
- `npm t` results in:

```
stderr | unknown test
[Vitest] takeCoverage is not available in this NodeJs version.
Coverage could be incomplete. Update to NodeJs 14.18.

stderr | unknown test
[Vitest] takeCoverage is not available in this NodeJs version.
Coverage could be incomplete. Update to NodeJs 14.18.
```

Add .nvmrc with v16?

-

```
❯ npm run test:ui                                                                                                                                                                                                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                                                                                                                                                                          
> itk-viewer-transfer-function-editor@0.0.1 test:ui /home/matt/src/itk-viewer-transfer-function-editor                                                                                                                                                                                                                                                                                                                                    
> vitest --ui                                                                                                                                                                                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                                                                                                                                          
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯                                                                                                                                                                                                                      
Error: The following dependencies are imported but could not be resolved:                                                                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                                                                                                                                          
  itk-viewer-reference-ui/src/referenceUIMachineOptions.js (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/itk-vtk-viewer/main.js)                                                                                                                                                                                                                                                                               
  @kitware/vtk.js/favicon (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                                                         
  @kitware/vtk.js/macros (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/itk-vtk-viewer/main.js)                                                                                                                                                                                                                                                                                                                 
  @kitware/vtk.js/Rendering/Profiles/Volume (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                                       
  @kitware/vtk.js/Rendering/Core/ColorTransferFunction (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                            
  @kitware/vtk.js/Rendering/Misc/FullScreenRenderWindow (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                           
  @kitware/vtk.js/IO/Core/HttpDataSetReader (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                                       
  @kitware/vtk.js/Common/DataModel/PiecewiseFunction (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                              
  @kitware/vtk.js/Rendering/Core/Volume (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                                           
  @kitware/vtk.js/Rendering/Core/VolumeMapper (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                                     
  @kitware/vtk.js/IO/Core/DataAccessHelper/HttpDataAccessHelper (imported by /home/matt/src/itk-viewer-transfer-function-editor/examples/vtk-js/VolumeMapperExample.js)                                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                                                                                                                                          
Are they installed?                                                                                                                                                                                                                                                                                                                                                                                                                       
    at optimizeDeps (/home/matt/src/itk-viewer-transfer-function-editor/node_modules/vite/dist/node/chunks/dep-9c153816.js:55767:15)                                                                                                                                                                                                                                                                                                      
    at processTicksAndRejections (internal/process/task_queues.js:95:5)                                                                                                                                                                                                                                                                                                                                                                   
    at async runOptimize (/home/matt/src/itk-viewer-transfer-function-editor/node_modules/vite/dist/node/chunks/dep-9c153816.js:56647:44)                                                                                                                                                                                                                                                                                                 
    at async Server.httpServer.listen (/home/matt/src/itk-viewer-transfer-function-editor/node_modules/vite/dist/node/chunks/dep-9c153816.js:56662:21)                                                                                                                                                                                                                                                                                    
     ```

- `npm run lint` has warnings related to the bundles.

**Comment by PaulHax** [comment]
Thanks for the thorough tire kick.  Addressed the issues, except:

> The itk-vtk-viewer example is not using this transfer function editor. This is a priority.

The editor is hard to see.  I added a border.  Is that what you mean?

image

**Comment by thewtex** [comment]
> . Addressed the issues,

Thank you!

> The editor is hard to see. I added a border. Is that what you mean?

Yes, I did not see it. Can it replace the existing vtkItkPiecewiseGaussianWidget instead of being placed below?

**Comment by PaulHax** [comment]
@thewtex The vtkItkPiecewiseGaussianWidget has been completely removed from the itk-vtk-viewer example.

File: examples/basic/main.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const app = document.querySelector<HTMLDivElement>('#app')!
+
+app.innerHTML = `
+  <div id='editorHome'></div>
+`
+
+const editorHome = document.querySelector<HTMLDivElement>('#editorHome')
+if (editorHome) {
+  const editor = new TransferFunctionEditor(editorHome)
+  console.log('Control points', editor.getPoints())
```
Comment (thewtex) [inline examples/basic/main.ts:18]: debug code

File: README.md [hunk 3]
```diff
@@ -1,3 +1,17 @@
 # itk-viewer-transfer-function-editor
 
 [![CI](https://github.com/InsightSoftwareConsortium/itk-viewer-transfer-function-editor/actions/workflows/ci.yml/badge.svg)](https://github.com/InsightSoftwareConsortium/itk-viewer-transfer-function-editor/actions/workflows/ci.yml)
+
+## Run examples
+
+```
+cd examples/itk-vtk-viewer
+npm ci
```
Comment (thewtex) [inline README.md:9]: needs a package-lock.json\

**Review by thewtex — APPROVED** [review]
@PaulHax awesome!!!

A few minor issues noted inline and good to merge!

File: README.md (hunk 3 above)
Comment (PaulHax) [reply README.md:9]: Updated.

**Pull request merged**